venom/rpc/reflect/service.py

```diff
@@ -3,31 +3,28 @@
 from venom.rpc.reflect.openapi import make_openapi_schema
 from venom.rpc.reflect.reflect import Reflect
 from venom.rpc.reflect.stubs import ReflectStub, OpenAPISchema
-from venom.rpc.service import ServiceManager
-from venom.util import MetaDict
 
 
-class ReflectServiceManager(ServiceManager):
-    def __init__(self, meta: MetaDict, meta_changes: MetaDict):
-        super().__init__(meta, meta_changes)
-        self.reflect = Reflect()
-        # TODO setup
+class ReflectService(Service):
+    class Meta:
+        stub = ReflectStub
 
-    def reflect_service(self, sender: Venom, service: Service):
-        self.reflect.add(service)
-        pass  # TODO add reflection (venom specific)
+    def __init__(self, venom: 'venom.rpc.Venom' = None) -> None:
+        super().__init__(venom)
+        self.reflect = Reflect()
 
-    def register(self, venom: 'venom.Venom'):
         for service in venom:
-            self.reflect_service(venom, service)
-        Venom.on_add_public_service.connect(self.reflect_service, sender=venom)
+            self._reflect_service(venom, service)
 
+        Venom.on_add_public_service.connect(self._reflect_service, sender=venom)
 
-class ReflectService(Service):
-    class Meta:
-        stub = ReflectStub
-        manager = ReflectServiceManager
+    def _reflect_service(self, sender: Venom, service: Service):
+        self.reflect.add(service)
+        for attr in sender.options:
```

Comment:
This is not good. The options could be anything, so you shouldn't just blindly copy all attributes from it. Also, you have access to venom in __init__, so this is not the right place to access venom. Also, those properties could be filled when Reflect is initialized.